starter-project-backend-g33/src/common/dal.ts

```diff
@@ -0,0 +1,119 @@
+import mongoose, { Model } from 'mongoose'
+
+
+
+const getOne = (
+    model:
+        Model<any, {}, {}>
+) => async (id: any) => {
+
+    try {
+        const doc = await model.findOne({ _id: id })
+        return doc
+    } catch (e) {
+        console.log(e)
+    }
```

Comment:
Let's remove the try and catch for all the DAL methods. The errorHandler middleware will take care of all our errors by trickling it down to all the req, and res as it is set in the app.js as the last middleware in the stack of middleware.

something like:
``` return await model.findOne({ _id: query }).select('-hash').exec() ```

Reply:
aight @GedionT done.